### ISSUE TITLE
Message from search is not shown in chat

### ISSUE THREAD

**Issue by mahibi** [nextcloud/talk-android#3527]
If search returns a message that is not yet loaded by adapter, it just won't show the message (and no info that it failed).

```
    private fun scrollToMessageWithId(messageId: String) {
        val position = adapter?.items?.indexOfFirst {
            it.item is ChatMessage && (it.item as ChatMessage).id == messageId
        }
        if (position != null && position >= 0) {
            binding.messagesListView.smoothScrollToPosition(position)
        } else {
            // TODO show error that we don't have that message?
        }
    }
```

https://github.com/nextcloud/talk-android/blob/59e8f4b550501b4ab833c44fee826829b8fa62dd/app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt#L2587C64-L2587C64

This was never implemented properly, so it exists since the search feature was added.
It maybe makes sense to even wait with this until offline support is implemented.

## reproduce
search for quite old message

## expected
scrolled to message

## actual
chat opens with latest messages. searched message is not shown.

**Comment by flop25** [comment]
Hello
I have 2 Questions about that:
its seems to be fixed in issue:4433 and https://github.com/nextcloud/talk-android/pull/4647 right?
And that fix is targeted to 21.1.0: when (approximately) will it be released, if you have an idea?

That lack/issue is a killer for my friends and family that I persuaded to use Talk instead of a corporate centralized messaging platform
because for them having a working search feature is a basic need excepted for a messaging app.

**Comment by mahibi** [comment]
yes, this was solved with https://github.com/nextcloud/talk-android/pull/4647 which will be released in first week of june.

But i see this is "only" solved for the main search in the conversations list.
It's not yet solved when searching via the upper right corner menu in a chat. 
I also see it's not implemented in iOS to search when being in a chat.

Is there a reason for this @rapterjet2004 @Ivansss @SystemKeeper ?

**Comment by Ivansss** [comment]
There is no search functionality whitin a conversation in the Talk iOS app yet, only the general search from conversation list.

**Comment by mahibi** [comment]
This will be solved with https://github.com/nextcloud/talk-android/pull/4889/commits/971cc79b76ea2b784cd7ae971a54130e832f2d75
